### PR TITLE
PRMDR-0000: Add consistency to handlers and utilise lambda exception decorator

### PR DESCRIPTION
**Pull request by abbas-khan10**

### CODE REVIEW

File: lambdas/services/document_deletion_service.py [hunk 1]
```diff
@@ -37,21 +40,25 @@ def delete_specific_doc_type(
...
+                type_of_delete=str(S3LifecycleTags.SOFT_DELETE.value),
+            )
+
+            logger.info(
+                f"Deleted document of type {doc_type.value}",
+                {"Result": "Successful deletion"},
+            )
+            return results
+        except (ClientError, DynamoServiceException) as e:
+            logger.error(f"Failed to delete documents: {str(e)}")
```
Comment (NogaNHS) [inline lambdas/services/document_deletion_service.py:63]: Can we add the results part `'"Result": f"Unsuccessful deletion due to {str(e)}"}'` to the log here?

**Review by NogaNHS — COMMENTED** [review]
We need to log those 'Result: XXX' for the exceptions somewhere. It was part of the logging requirement. We can add those to the service when we raise the Error.

File: lambdas/handlers/login_redirect_handler.py [hunk 2]
```diff
@@ -14,18 +14,11 @@
 @set_request_context_for_logging
 @override_error_check
 @ensure_environment_variables(names=["AUTH_DYNAMODB_NAME"])
+@handle_lambda_exceptions
 def lambda_handler(event, context):
     request_context.app_interaction = LoggingAppInteraction.LOGIN.value
     login_redirect_service = LoginRedirectService()
```
Comment (joefong-nhs) [inline lambdas/handlers/login_redirect_handler.py:20]: Suggest to add an entry point log just for consistency

```suggestion
    login_redirect_service = LoginRedirectService()
    logger.info("Login Redirect handler triggered")
```

File: lambdas/handlers/search_patient_details_handler.py [hunk 3]
```diff
@@ -14,32 +15,23 @@
 @set_request_context_for_logging
 @validate_patient_id
 @override_error_check
+@handle_lambda_exceptions
 def lambda_handler(event, context):
     request_context.app_interaction = LoggingAppInteraction.PATIENT_SEARCH.value
+    nhs_number = event["queryStringParameters"]["patientId"]
+    request_context.patient_nhs_no = nhs_number
```
Comment (joefong-nhs) [inline lambdas/handlers/search_patient_details_handler.py:22]: Suggest to add an entry point log just for consistency

```suggestion
    request_context.patient_nhs_no = nhs_number
    logger.info("Starting document reference search process")
```

File: lambdas/handlers/token_handler.py [hunk 4]
```diff
@@ -18,6 +19,7 @@
 @ensure_environment_variables(
     names=["AUTH_STATE_TABLE_NAME", "AUTH_SESSION_TABLE_NAME"]
 )
+@handle_lambda_exceptions
 def lambda_handler(event, context):
     request_context.app_interaction = LoggingAppInteraction.LOGIN.value
```
Comment (joefong-nhs) [inline lambdas/handlers/token_handler.py:25]: Suggest to add an entry point log just for consistency

```suggestion
    logger.info("Token request handler triggered")
```

File: lambdas/services/base/dynamo_service.py [hunk 5]
```diff
@@ -85,8 +84,7 @@ def query_with_requested_fields(
 
             return results
         except ClientError as e:
-            logger.error(f"Unable to query table: {table_name}")
-            logger.error(e)
+            logger.error(e, {"Result": "Unable to query table: {table_name}"})
```
Comment (joefong-nhs) [inline lambdas/services/base/dynamo_service.py:87]:
```suggestion
            logger.error(e, {"Result": f"Unable to query table: {table_name}"})
```

**Review by joefong-nhs — APPROVED** [review]

File: lambdas/services/pds_api_service.py [hunk 6]
```diff
@@ -48,7 +48,7 @@ def pds_request(self, nhs_number: str, retry_on_expired: bool):
             return pds_response
 
         except (ClientError, JSONDecodeError) as e:
-            logger.error(f"Error when getting ssm parameters {e}")
+            logger.error(str(e), {"Results": "Error when getting ssm parameters"})
```
Comment (NogaNHS) [inline lambdas/services/pds_api_service.py:51]: Should be result not result**s**

**Review by NogaNHS — APPROVED** [review]

**Pull request merged**